From fd9209c3ea45def87bda9d0b2411d0a701ed38a5 Mon Sep 17 00:00:00 2001
From: Jin Yao <yao.jin@linux.intel.com>
Date: Mon, 12 May 2014 20:02:58 +0800
Subject: [PATCH 123/152] Change DW DMA suspend_noirq and resume_noirq
 callbacks to suspend_late and resume_early

we saw suspend process hang at here and never resume.

[   60.595667] PM: device INTL9C60:01 driver [dw_dmac] suspend noirq enter

Since DMA services for other components, we need to ensure DMA suspends
after the components and resume before the components are active.

Signed-off-by: Jin Yao <yao.jin@linux.intel.com>
---
 drivers/dma/dw/platform.c |    8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/drivers/dma/dw/platform.c b/drivers/dma/dw/platform.c
index 91c89cd..c872392 100644
--- a/drivers/dma/dw/platform.c
+++ b/drivers/dma/dw/platform.c
@@ -283,12 +283,8 @@ static int dw_resume_noirq(struct device *dev)
 #endif /* !CONFIG_PM_SLEEP */
 
 static const struct dev_pm_ops dw_dev_pm_ops = {
-	.suspend_noirq = dw_suspend_noirq,
-	.resume_noirq = dw_resume_noirq,
-	.freeze_noirq = dw_suspend_noirq,
-	.thaw_noirq = dw_resume_noirq,
-	.restore_noirq = dw_resume_noirq,
-	.poweroff_noirq = dw_suspend_noirq,
+	.suspend_late = dw_suspend_noirq,
+	.resume_early = dw_resume_noirq,
 };
 
 static struct platform_driver dw_driver = {
-- 
1.7.10.4

